RNA-seq/05-nb_cell_line_DESeq2.Rmd

```diff
@@ -154,124 +169,131 @@ Let's save this to our results file.
 readr::write_rds(deseq_object, file = deseq_file)
 ```
 
-Let's take a look at the results table.
+Now we will have a look at the results table.
 
 ```{r deseq_results}
 deseq_results <- results(deseq_object)
 deseq_results
 ```
 
-```{r arrange-deseq_results}
-# this is of class DESeqResults -- we want a data.frame
-deseq_df <- deseq_results %>%
-  # make into data.frame
-  as.data.frame() %>%
-  # the gene names are rownames -- let's make this it's own column for easy 
-  # display
-  tibble::rownames_to_column(var = "Gene")
-
-deseq_df %>%
-  # let's sort by statistic -- the highest values should be what is up in the
-  # MYCN amplified cell lines
-  dplyr::arrange(dplyr::desc(stat))
+How many genes were differentially expressed (FDR < 0.05)?
+
+```{r results_summary}
+summary(deseq_results, alpha = 0.05)
 ```
 
-Let's take a look at what the top gene is: [`ENSG00000134323`](http://useast.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000134323)
 
-How many genes were differentially expressed (FDR < 0.05)?
+#### Shrinking log2 fold change estimates
 
-```{r summary}
-summary(results(deseq_object), alpha = 0.05)
-```
+The estimates of log2 fold change calculated by `DESeq()` are not corrected for expression level.
+This means that when counts are small, we are likely to end up with some large fold change values that overestimate the true extent of the change between conditions.
 
-**Write full data.frame to file**
+We can correct this by applying a "shrinkage" procedure, which will adjust large values with small counts downward, while preserving values where we have larger counts, which are likely to be more accurate.
 
-```{r write_tsv}
-readr::write_tsv(deseq_df, file = deseq_df_file)
+To do this, we will use the `lfcShrink()` function, but first we need to know the name and/or position of the "coefficient" that was calculated by `DESeq()`, which we can do with the `resultsNames()` function
+
+```{r deseq_coef}
+# get the deseq coefficient names:
+resultsNames(deseq_object)
 ```
 
-#### Visualize differentially expressed genes with a heatmap
+The figure we are interested in is the status measure, which is in position 2.
 
-Let's get an overall picture of our results using a heatmap. 
-Before we can do this, we will need to transform our data so that when we make the heatmap the genes are on a more similar scale and the heatmap is interpretable. 
-We will use variance stabilizing transformation from DESeq2 to do this.
-Refer to the help page or the DESeq2 docs for more information. 
+There are a few options for the shrinkage estimation. 
+The default is `apeglm` ([Zhu _et al._ 2018](https://doi.org/10.1093/bioinformatics/bty895)), but we have found that this can be sensitive to extreme outliers, which are definitely a factor in this data set.
```

Comment:
Do we want to generally encourage them to still try `apeglm` first?

Reply:
I am agnostic to this question, which I why I didn't address it directly. Obviously the authors think you should use `apeglm` and it is "more principled" in some ways (it is specifically designed for expression count data, where `ashr` is much more general) but I don't feel like I really have enough of a sense to make a definitive recommendation. We have both installed, so people can try them and see what happens if it pleases them. We might ask them to do that in the exercises?